ouroboros-consensus/src/Ouroboros/Consensus/Storage/ChainDB/Impl/ChainSel.hs

```diff
@@ -714,16 +717,10 @@ chainSelectionForBlock cdb@CDB{..} blockCache hdr punish = do
          -- ^ Chain and ledger to switch to
       -> StrictTVar m (StrictMaybe (Header blk))
          -- ^ Tentative header
-      -> (    [LedgerEvent blk]
-           -> NewTipInfo blk
-           -> AnchoredFragment (Header blk)
-           -> AnchoredFragment (Header blk)
-           -> TraceAddBlockEvent blk
-         )
-         -- ^ Given the 'NewTipInfo', the previous chain, and the new chain,
-         -- return the event to trace when we switched to the new chain.
+      -> ChainSwitchType
+         -- ^ Whether we are extending the current chain or switching to a fork.
```

Comment:
I think this comment would be better suited as the Haddock on `ChainSwitchType`.